### PR TITLE
Cron scheduler api

### PR DESCRIPTION
**Pull request by eryshev**
MVP of Cron Scheduler API.
    
    Main features:
    - CronApp class with minimalistic public and private API was created,
    supported API is listed under https://documenter.getpostman.com/view/3548308/RznCs12X
    Briefly we can see a status of a job, list job executions, see execution's logs and
    pause/resume a job.
    - CronProject takes care of instantiation of db connection, executor, scheduler
    and CronApp API server
    - HelloCronScheduling example was updated accordingly, check it out!
    
    Others:
    - introducing thread pool for Cron
    - sse method moved to global cuttle utils package object
    - state of CronScheduler became a separate class with methods which allow atomic mutation of 
    underlying stm
    - some scalafmt was applied to TimeSeriesApp

### CODE REVIEW

File: core/src/main/scala/com/criteo/cuttle/ThreadPools.scala [hunk 1]
```diff
@@ -32,6 +31,8 @@ object ThreadPools {
 
   sealed trait DoobieThreadPool extends WrappedThreadPool with Metrics
 
+  sealed trait CronThreadPool extends WrappedThreadPool with Metrics
```
[the remark this answers is not in the document]
Comment (eryshev) [reply core/src/main/scala/com/criteo/cuttle/ThreadPools.scala]: I moved it in cron package but to do it I had to remove sealed from WrappedThreadPool and Metrics.

File: cron/src/main/scala/com/criteo/cuttle/cron/CronApp.scala [hunk 2]
```diff
@@ -0,0 +1,152 @@
+package com.criteo.cuttle.cron
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+import scala.util.{Try, _}
```
[the remark this answers is not in the document]
Comment (eryshev) [reply cron/src/main/scala/com/criteo/cuttle/cron/CronApp.scala]: I'm not sure why you're thinking it's not used

**Comment by eryshev** [comment]
@dufrannea comments were applied

**Comment by guillaumebort** [comment]
It looks like we are missing endpoint to get the actual scheduler state?

**Comment by eryshev** [comment]
> It looks like we are missing endpoint to get the actual scheduler state?

What about [/api/dashboard](https://documenter.getpostman.com/view/3548308/RznCs12X#431c0ac3-e59b-419a-9e41-dd19654d8407)

**Pull request merged**